### ISSUE TITLE
Errors that occur in crud operations made through AppServices are not displayed as a message on the screen.

### ISSUE THREAD

**Issue by akilliadam** [abpframework/abp#9910]
On the pages where we inject the AppServices we have created, Abp framework does not catch the errors that may occur in the crud operations we perform through these services and do not display them as a message on the screen. Errors occur on the console screen and the program is stopped. In order to display the resulting error messages on the screen, the relevant page must be inherited from AbpComponentBase, as stated in the Abp Framework documents. This is not very useful. How can we catch and display the errors that may occur as a result of crud operations on a service we injected, without writing extra code (such as try-catch). Is there a different way? Or do you have any plans for this situation in the future?

**Comment by maliming** [comment]
hi

* Your **ABP Framework version**.
* Your **User Interface** type (Angular/MVC/React... etc.) if the issue is related to a specific UI
* Your database provider(EF Core/MongoDB)
* Exception message and **stack trace** if available (check the logs).
* Steps needed to **reproduce** the problem.

**Comment by akilliadam** [comment]
ABP Framework version : 4.4 (latest)
User Interface : Blazor Server
DbProvider : EF Core


I will give the Authors Page in the BookStore project as an example.
I set the project type to WebAssembly. I have injected IAuthorAppService in Authors Page. Later on this page AuthorAppService.GetAsync(), AuthorAppService.UpdateAsync() etc. When I use crud operations, if any error occurs during these operations, the errors that occur are shown on the screen for the messagebox without any problems. However, when I change the project type to BlazorServer, if an error occurs during crud operations, the program breaks and error messages are displayed on the console.

This situation is stated in Abp's documentation and it is stated that we should use HandleErrorAsync in catch block using try catch for crud operations to show error messages.

This method is not useful at all. In other words, I need to be able to use crud operations through the IAuthorAppService that I inject and display the error messages on the screen. Is there any other way you can suggest for this situation? Or can we not do that in Blazor Server. @maliming

**Comment by akilliadam** [comment]
@maliming can you answer me?

**Comment by maliming** [comment]
hi

Steps needed to reproduce the problem.

**Comment by akilliadam** [comment]
I am using Acme.BookStore project which is your Blazor WebAssembly project available on github.
I run the project without any changes. Then I list the authors. I then open any of the authors for editing.
I'm making changes to the author information, but I haven't saved it yet.
By switching to the database side, I manually delete the author from the database.
Then I want to save the author by switching to the browser.
Since I deleted the author from the database, it shows an error message that there is no author with ..... Id in the messagebox as in the picture below. This is what has to happen. This way the project runs smoothly.

![BlazorWbAssemblyError](https://user-images.githubusercontent.com/80377964/132201590-e21d128c-ee5f-4d5e-911a-e1cda3080212.PNG)

I then create a Blazor Server project. I am using the same codes in this project without changing them at all.
Likewise, I open the authors list first. I'm opening to replace one of the existing authors. I manually delete the author I opened from the database.
Then I switch to the browser and I want to save by making a few changes to the author information.
Since the author has been deleted from the database, I was expecting it to show the author not found error message with .....Id in the message box as shown in the picture above. But the project was broken and it showed this error message in Browser's console as in the image below.

![BlazorServerError](https://user-images.githubusercontent.com/80377964/132204075-2029aeb2-1fb1-4526-9cde-618703d88332.PNG)

When I examine the documents of Abp, we need to inherit the relevant page from the AbpComponentBase component in order to show the error messages in the Message box in Balzor Server projects.
It is stated that we should use a function similar to the one below to catch the error messages that occur after inheritance.
When you do it this way, the project runs smoothly again and shows error messages in the message box on the screen.

```
protected async Task<TGetOutputDto> CreateAsync(TCreateInput input)
        {
            try
            {
                return await CrudService.CreateAsync(input);
            }
            catch (Exception e)
            {
                await HandleErrorAsync(e);
            }

            return default;
        }
```

However, this method is not useful at all. While I can use AuthorAppService.CreateAsync(), AuthorAppService.DeleteAsync(), AuthorAppService.UpdateAsync() directly in Blazor WebAssembly projects, I need to use these functions in a try catch to catch errors in Blazor Server projects.
Is there any other way to do this without using try catch like in Blazor WebAssembly? If so, how should I do it?
Thanks. @maliming
